### PR TITLE
Add support for underscores in DYNAMIC_SIDECAR_DOCKER_IMAGE_RE

### PR DESCRIPTION
**Pull request by mrnicegyu11**
Our release workflow creates image tags with underscores in the name (see for example https://hub.docker.com/layers/itisfoundation/dynamic-sidecar/staging-github-staging_diolkos1-2022-06-15--15-04.75ddf7e3fb86944ef95fcf77e4075464848121f1/images/sha256-16bef83f05d3d98c04d901700e6001358441feb2eeb2b5c144530ac7c838c05e?context=explore ). For a proper rollback, these older images of the dynamic-sidecar need to be able to pass the regex check in the director-v2, so they can be used.

Check also here: https://regex101.com/r/TH33ZS/1

### CODE REVIEW

File: services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # - itisfoundation
 # - 10.0.0.0:8473 (IP & Port)
 DYNAMIC_SIDECAR_DOCKER_IMAGE_RE = (
-    r"(^([a-zA-Z0-9:.-]+)/)?(dynamic-sidecar):([a-zA-Z0-9.-]+$)"
+    r"(^([_a-zA-Z0-9:.-]+)/)?(dynamic-sidecar):([_a-zA-Z0-9.-]+$)"
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py:15]: Please run the test i attached with and without this chanve and add more string samples

**Review by pcrespov — APPROVED** [review]
I added a test. 
Could you please add more use-cases?

File: services/director-v2/tests/unit/test_models_schemas_constants.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    # TODO: add test with REGEX_DY_SERVICE_PROXY,
+    # TODO: add test with  REGEX_DY_SERVICE_SIDECAR,
+    DYNAMIC_SIDECAR_DOCKER_IMAGE_RE,
+)
+
+
+@pytest.mark.parametrize(
+    "sample",
+    [
+        "itisfoundation/dynamic-sidecar:staging-github-staging_diolkos1-2022-06-15--15-04.75ddf7e3fb86944ef95fcf77e4075464848121f1",
```
Comment (sanderegg) [inline services/director-v2/tests/unit/test_models_schemas_constants.py:18]: and maybe add as parameter the release tag version as well... just to be on the safe side

**Review by sanderegg — APPROVED** [review]

**Comment by pcrespov** [comment]
@mrnicegyu11 do we still need this?

**Pull request merged**